lxd/instance/drivers/driver_qemu.go

```diff
@@ -1267,6 +1270,144 @@ echo "To start it now, unmount this filesystem and run: systemctl start lxd-agen
 		return err
 	}
 
+	// Templated files.
+	err = os.MkdirAll(filepath.Join(configDrivePath, "files"), 0500)
+	if err != nil {
+		return err
+	}
+
+	// Template anything that needs templating.
+	key := "volatile.apply_template"
+	if vm.localConfig[key] != "" {
+		// Run any template that needs running.
+		err = vm.templateApplyNow(vm.localConfig[key], filepath.Join(configDrivePath, "files"))
+		if err != nil {
+			return err
+		}
+
+		// Remove the volatile key from the DB.
+		err := vm.state.Cluster.ContainerConfigRemove(vm.id, key)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = vm.templateApplyNow("start", filepath.Join(configDrivePath, "files"))
```

Comment:
@stgraber is there a reason we load templates from the `files` dir rather than calling it `templates`? is that historical?

Reply:
Because they're not templates at this stage anymore, they're rendered copies, so "files" felt more accurate.